subgraph-js/src/buildSubgraphSchema.ts

```diff
@@ -30,6 +32,23 @@ type LegacySchemaModule = {
 
 export { GraphQLSchemaModule };
 
+function missingFederationDirectives(modules: GraphQLSchemaModule[]): GraphQLDirective[] {
+  // Copying the array as we're going to modify it.
+  const missingDirectives = federationDirectives.concat();
```

Comment:
May not matter too much, but it would probably be better to use a set to avoid the O(n) lookup in a loop.